oci/utils/tarball.go

```diff
@@ -0,0 +1,41 @@
+package utils
+
+import (
+	"archive/tar"
+	"fmt"
+	"io"
+	"io/fs"
+	"path/filepath"
+)
+
+// OpenFromTarOCILayer returns a reader for the given file from the provided
+// tarball reader.
+func OpenFromTarOCILayer(r io.Reader, path string) (io.Reader, error) {
+	tarPath, err := filepath.Rel("/", path)
+	if err != nil {
+		return nil, fmt.Errorf("trimming leading separator from path %q: %w", path, err)
+	}
+
+	tr := tar.NewReader(r)
+	for {
+		h, err := tr.Next()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return nil, fmt.Errorf("advancing to the next entry in the OCI image layer: %w", err)
+		}
+
+		if h.Name != tarPath {
+			continue
+		}
+
+		if t := h.Typeflag; t != tar.TypeReg {
+			return nil, fmt.Errorf("file has wrong type %d", t)
```

Comment:
I think the file code/number might actually be more confusing than helpful, but it's up to you this one

Reply:
i will change it to:

```suggestion
			return nil, fmt.Errorf("path is not a regular file")

```